tests/test_simple.py

```diff
@@ -865,3 +873,94 @@ def test_del_contact(self) -> None:
             access_token=self.access_token,
         )
         assert channel.code == 404, channel.result
+
+
+class FederationDomainSchemaTest(ModuleApiTestCase):
+    """
+    Test that the required fields for the federation list are present and parsable.
+    See:
+    https://github.com/gematik/api-vzd/blob/main/src/schema/FederationList.json
+    for the schema to use.
+
+    As of the time of this writing, these are fields that are required:
+        domain: str
+        telematikID: str
+        timAnbieter: str
+        isInsurance: bool
+
+    """
+
+    def prepare(
+        self, reactor: MemoryReactor, clock: Clock, homeserver: HomeServer
+    ) -> None:
+        super().prepare(reactor, clock, homeserver)
+
+        self.invchecker = InviteChecker(
+            self.hs.config.modules.loaded_modules[0][1], self.hs.get_module_api()
+        )
+
+    async def extract_entry_from_domainList(
+        self, domain: str
+    ) -> FederationDomain | None:
+        """
+        Search for a specific domain in the federation list
+        """
+        _, raw_fed_list = await self.invchecker.fetch_federation_allow_list()
+        assert len(raw_fed_list.domainList) > 0
+
+        for domain_entry in raw_fed_list.domainList:
+            if domain_entry.domain == domain:
+                return domain_entry
+
+        return None
```

Comment:
You can probably just assert here instead of returning None

Reply:
I'm not certain what to `assert` here, and there may be a case for a test of what to do when a domain isn't present in the future. What do you recommend?

Reply:
Well, we are always expecting to find those items in the tests which call it. So just assert for now and change it, if we ever have tests which expect None. Gets rid of some None checks at the caller side and makes the return type easier to read as well.

Reply:
So just `assert False, f"Not found in federation list {domain}"`

Reply:
That sounds better than what I had in my head
fixed in e64ae8f